mutiny-core/src/fedimint.rs

```diff
@@ -0,0 +1,139 @@
+use crate::{error::MutinyError, logging::MutinyLogger, storage::MutinyStorage};
+use bitcoin::{secp256k1::Secp256k1, util::bip32::ExtendedPrivKey};
+use bitcoin::{
+    util::bip32::{ChildNumber, DerivationPath},
+    Network,
+};
+use fedimint_client::{derivable_secret::DerivableSecret, ClientArc, FederationInfo};
+use fedimint_core::db::mem_impl::MemDatabase;
+use fedimint_core::{api::InviteCode, config::FederationId};
+use fedimint_ln_client::LightningClientInit;
+use fedimint_mint_client::MintClientInit;
+use fedimint_wallet_client::WalletClientInit;
+use lightning::log_info;
+use lightning::util::logger::Logger;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    str::FromStr,
+    sync::{atomic::AtomicBool, Arc, RwLock},
+};
+
+const FEDIMINT_CLIENT_NONCE: &[u8] = b"Fedimint Client Salt";
+
+// This is the FedimintStorage object saved to the DB
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+pub struct FedimintStorage {
+    pub fedimints: HashMap<String, FedimintIndex>,
+    #[serde(default)]
+    pub version: u32,
+}
+
+// This is the FedimintIdentity that refer to a specific node
+// Used for public facing identification.
+pub struct FedimintIdentity {
+    pub uuid: String,
+    pub federation_id: FederationId,
+}
+
+// This is the FedimintIndex reference that is saved to the DB
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone, Default)]
+pub struct FedimintIndex {
+    pub child_index: u32,
+    pub federation_code: String,
+    pub archived: Option<bool>,
```

Comment:
dont think this needs to be optional, we only did that on node index for backwards compatibility